torch/csrc/distributed/c10d/ProcessGroupWrapper.cpp

```diff
@@ -62,18 +72,56 @@ struct CollectiveFingerPrint {
     verify_tensors(inp, pg);
   }
 
+  // Takes a serialized fingerprint from
+  // CollectiveFingerPrint::serialize_fingerprint and deserializes it back to a
+  // CollectiveFingerPrint struct
+  CollectiveFingerPrint deserialize_fingerprint(at::Tensor serialized_tensor) {
+    // TODO: Need to add asserts to validate serialized_tensor.sizes() before
+    // deserializing
+    int index = 0;
+    // 1. OpType
+    OpType optype = OpType(serialized_tensor[index].item<int>());
```

Comment:
might want to add some validation to ensure we don't access out of bounds and have a hard to debug error.